### ISSUE TITLE
Type onAdditionalDetails argument in CoreOptions

### ISSUE THREAD

**Issue by dimitriBouteille** [Adyen/adyen-web#2536]
The argument `state` is `any` actuality for `CoreOptions.onAdditionalDetails` : 

https://github.com/Adyen/adyen-web/blob/78bfdd54e4aeb35c085c9ed77ed6878fd5575cef/packages/lib/src/core/types.ts#L136

Maybe add type for this argument, something like that : 

```ts
export interface OnAdditionalDetailsData {
  data: {
    details: {
      [key: string]: unknown;
    };
  };
}
```

**Comment by ribeiroguilherme** [comment]
Hey @dimitriBouteille . 

We are fixing all typescript interfaces/types for the new major version that will come in the upcoming weeks. This includes fixing the type of this callback as well. I will keep this ticket open until we release the new major version containing its fix.

Cheers.

**Comment by ashrafnazar** [comment]
@ribeiroguilherme - Could you also please provide examples of how to use this as well?

Not sure if the example in the docs is correct:

```js
function handleOnAdditionalDetails(state, component) {
   state.data // Provides the data that you must pass in the `/payments/details` call.
   component // Provides the active component instance that called this event.
 }
```

TypeScript examples would be much appreciated as well

**Comment by ashrafnazar** [comment]
Hi @ribeiroguilherme - thanks for the reply. Here is the [link](https://docs.adyen.com/online-payments/3d-secure/native-3ds2/web-component/?tab=make-payment-own-form_2#handle-the-3d-secure-2-action). You'll have to scroll down a little bit.

Having an example in the documentation (or at least links) would be beneficial. I still think having TS examples would be beneficial, as types cannot always be inferred.

I haven't seen examples for the `checkout.createFromAction` method either.

**Comment by ribeiroguilherme** [comment]
I reached out to the Docs team and we will fix the `onAdditionalDetails` code sample in the next days.

Can you elaborate what do you expect to see about `checkout.createFromAction` ? 
It is a helper to create your 3DS component. As the documentation says: you pass the action from /payments response to it and optionally you can pass a  `threeDSConfiguration`. Then you `mount` it in the DOM node. After that, you should see the 3DS in your webpage.

**Comment by ashrafnazar** [comment]
@ribeiroguilherme - appreciate you reaching out.

As I'm working on a Next.js app, handling how to retrieve the `action` property from the `/payments` response would be good.

So what I'm having to do is use React's Context API to retrieve the property from the API call and then pass it into `createFromAction`. If you feel there is a better way to do it, then please expound on this.

**Comment by ribeiroguilherme** [comment]
Hey @dimitriBouteille - we released v6.0.0 which contains many improvements, one of which we improved the Typescript exports and we added types for different callbacks.

I recommend to try it out and reach out to us in case you notice/miss something. Cheers.
